### PR TITLE
Fix every instance of `unicorn/no-array-callback-reference`

### PR DESCRIPTION
**Pull request by darkred**
Re https://github.com/sindresorhus/refined-github/issues/4008#issuecomment-888167968

Based on the [<code>unicorn/no-array-callback-reference</code> documentation](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/no-array-callback-reference.md#pass) , 1st example:

```
Fail: const foo = array.map(callback);
Pass: const foo = array.map(element => callback(element));
```

## Test URLs

N/A

## Screenshot

N/A

### CODE REVIEW

File: .xo-config.json [hunk 1]
```diff
@@ -14,7 +14,7 @@
 		"@typescript-eslint/no-unsafe-member-access": "off",
 		"@typescript-eslint/no-unsafe-return": "off",
 		"@typescript-eslint/no-unsafe-call": "off",
-		"unicorn/no-array-callback-reference": "warn",
+		"unicorn/no-array-callback-reference": "error",
```
Comment (fregante) [inline .xo-config.json:17]: This is the default, so it can be dropped

**Review by fregante — CHANGES_REQUESTED** [review]
Not all of them are elements, they should reflect what they are

File: source/features/open-all-conversations.tsx [hunk 2]
```diff
@@ -27,7 +27,7 @@ function onButtonClick(): void {
 	}
 
 	void browser.runtime.sendMessage({
-		openUrls: issues.map(getUrlFromItem),
+		openUrls: issues.map(element => getUrlFromItem(element)),
```
Comment (fregante) [inline source/features/open-all-conversations.tsx:30]: issues -> issue
```suggestion
		openUrls: issues.map(issue => getUrlFromItem(issue)),
```

File: source/features/mark-merge-commits-in-list.tsx [hunk 3]
```diff
@@ -39,7 +39,7 @@ export function getCommitHash(commit: HTMLElement): string {
 
 async function init(): Promise<void> {
 	const pageCommits = select.all('li.js-commits-list-item');
-	const mergeCommits = await filterMergeCommits(pageCommits.map(getCommitHash));
+	const mergeCommits = await filterMergeCommits(pageCommits.map(element => getCommitHash(element)));
```
Comment (fregante) [inline source/features/mark-merge-commits-in-list.tsx:42]: pageCommits -> commit
```suggestion
	const mergeCommits = await filterMergeCommits(pageCommits.map(commit => getCommitHash(commit)));
```

File: source/features/list-prs-for-file.tsx [hunk 4]
```diff
@@ -129,7 +129,7 @@ async function init(): Promise<void> {
 							<>
 								Careful, {prs.length} open PRs are already touching this file
 								<span className="ml-2 BtnGroup" style={{verticalAlign: '-0.6em'}}>
-									{prs.map(getSingleButton)}
+									{prs.map(element => getSingleButton(element))}
```
Comment (fregante) [inline source/features/list-prs-for-file.tsx:132]: prs -> pr 
```suggestion
									{prs.map(pr => getSingleButton(pr))}
```

**Review by fregante — COMMENTED** [review]
Examples

File: .xo-config.json (hunk 1 above)
Comment (darkred) [reply .xo-config.json:17]: Thanks.

File: source/features/reactions-avatars.tsx [hunk 5]
```diff
@@ -54,7 +54,7 @@ async function showAvatarsOn(commentReactions: Element): Promise<void> {
 
 	const participantByReaction = select
 		.all(':scope > button', commentReactions)
-		.map(getParticipants);
+		.map(element => getParticipants(element));
```
Comment (darkred) [inline source/features/reactions-avatars.tsx:57]: I kept the variable as element (1/2)

File: source/features/tag-changelog-link.tsx [hunk 6]
```diff
@@ -89,7 +89,7 @@ async function init(): Promise<void> {
 	// Look for tags in the current page and the next page
 	const pages = [document, await getNextPage()];
 	await domLoaded;
-	const allTags = select.all(tagsSelector, pages).map(parseTags);
+	const allTags = select.all(tagsSelector, pages).map(element => parseTags(element));
```
Comment (darkred) [inline source/features/tag-changelog-link.tsx:92]: I kept the variable as element (2/2)

File: source/features/reactions-avatars.tsx (hunk 5 above)
Comment (fregante) [reply source/features/reactions-avatars.tsx:57]: These are buttons. We can also change the signature of the callback from "container: HTMLElement" to "button: HTMLButtonElement" I think

File: source/features/tag-changelog-link.tsx (hunk 6 above)
Comment (fregante) [reply source/features/tag-changelog-link.tsx:92]: tag

Comment (darkred) [reply source/features/tag-changelog-link.tsx]: Ok, done.

File: source/features/reactions-avatars.tsx (hunk 5 above)
Comment (darkred) [reply source/features/reactions-avatars.tsx]: Both done (regarding the latter, I think I did it ok).

Comment (darkred) [reply source/features/reactions-avatars.tsx]: Sorry, I missed `HTMLButtonElement`. I'll revert and do a new commit.

Comment (darkred) [reply source/features/reactions-avatars.tsx]: Done.

**Comment by fregante** [comment]
Great, thanks!

**Pull request merged**